nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java

```diff
@@ -1122,7 +1123,7 @@ public long contentLength() {
 
         final ComponentLog logger = getLogger();
         for (String headerKey : dynamicPropertyNames) {
-            String headerValue = context.getProperty(headerKey).evaluateAttributeExpressions(requestFlowFile).getValue();
+            String headerValue = StringEscapeUtils.escapeJava(context.getProperty(headerKey).evaluateAttributeExpressions(requestFlowFile).getValue());
```

Comment:
Doesn't this assume that the server side has to unescape it? I'd rather use OkHttp's mechanism for not validating header values, but that assumes a later version and there are a whole set of Jira cases around the upgrade path, due to security issues, breaking changes, etc.

Maybe instead we could add (un)escapeJava methods to Expression Language, then the user can use those functions in EL to escape the values as they choose.

Reply:
I'm new in NIFI and I don't know it very well, I just open it because I'm developing a new custom processor for my personal purpose that use invokehttp as base and in my context I thought it was the best solution. I was thought it was a bug but you are right. Thank you for the explanation

Reply:
[RFC 2616 Section 2.2](https://tools.ietf.org/html/rfc2616#section-2.2) indicates that text values, used in HTTP headers, may contain characters other than ISO-8859-1 only when encoding according to [RFC 2047](https://tools.ietf.org/html/rfc2047).  Support for encoded values would depend on the receiving server implementation as @mattyb149 noted.  Implementing a solution inside the Expression Language context would provide greater flexibility for use cases where supporting characters outside of ISO-8859-1 in HTTP headers is required.

Reply:
@exceptionfactory thank you for the explanation too.